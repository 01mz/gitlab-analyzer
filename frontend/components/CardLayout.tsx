import {Box, Typography} from "@material-ui/core";
import React from "react";
import Image from "next/image";
import AppGradientBackground from "./AppGradientBackground";
import ChildrenProps from "../interfaces/ChildrenProps";

type CardLayoutProps =  ChildrenProps & {
    size?: "sm" | "md"
    smLogo?: boolean
}

const CardLayout = ({children, size, smLogo}: CardLayoutProps) => {
    // Change card width based on size prop
    let width = "60vw";
    width = size == "sm" ? "20vw": width;
    width = size == "md" ? "60vw": width;

    let logoSize = smLogo ? "50": "80";

    return (
        <AppGradientBackground>
            <Box
                bgcolor="primary.contrastText"
                boxShadow={20}
                width={width}
                height="60vh"
                minWidth="250px"
                minHeight="450px"
                borderRadius={45}
                padding="20px"
                display="flex"
                flexDirection="column"
                justifyContent="center"
            >
                <Box id="card-header" margin="10px" textAlign="center">
                    <Image
                        src="/gitlab.svg"
                        alt="The Gitlab Logo"
                        width={logoSize}
                        height={logoSize}
                    />
                    <Typography variant="h6">
                        Gitlab Analyzer
                    </Typography>
                </Box>
                {children}
            </Box>
        </AppGradientBackground>
    );
};

export default CardLayout;
