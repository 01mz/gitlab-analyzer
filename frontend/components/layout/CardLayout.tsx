import {Box, Typography, Link, Icon} from "@material-ui/core";
import React from "react";
import Image from "next/image";
import AppGradientBackground from "../app/AppGradientBackground";
import ChildrenProps from "../../interfaces/ChildrenProps";
import NextLink from 'next/link'
import {makeStyles} from "@material-ui/styles";

type CardLayoutProps =  ChildrenProps & {
    size?: "sm" | "md"
    logoType?: "center" | "header"
    backLink?: string
    backLabel?: string
}

const useStyles = makeStyles({
    linkBack: {
        display: 'flex',
        alignItems: 'center',
        flexWrap: 'wrap'
    }
})

const CardLayout = ({children, size, logoType = "center", backLink, backLabel}: CardLayoutProps) => {
    // Change card width based on size prop
    let width = "60vw";
    width = size == "sm" ? "20vw": width;
    width = size == "md" ? "60vw": width;

    let logoSize = logoType === "center" ? "120" : "80" ;
    const classes = useStyles();

    return (
        <AppGradientBackground>
            <Box
                bgcolor="primary.contrastText"
                boxShadow={20}
                width={width}
                minWidth="250px"
                minHeight="450px"
                borderRadius={45}
                padding="20px"
                display="flex"
                flexDirection="column"
                justifyContent="center"
            >
                <Box id="card-header" margin="10px" textAlign="center">
                    <Image
                        src="/gitlab.svg"
                        alt="The Gitlab Logo"
                        width={logoSize}
                        height={logoSize}
                    />
                    <Typography variant="h6">
                        Gitlab Analyzer
                    </Typography>
                </Box>
                <Box
                    flex= {logoType === "center" ? "none" : 1}
                    display="flex"
                    flexDirection="column"
                    justifyContent="center"
                >
                    {children}
                </Box>
                {backLink &&
                    <Box>
                        <NextLink href={backLink} passHref>
                            <Link classes={{root:classes.linkBack}}>
                                <Icon fontSize="small">arrow_back</Icon>
                                <Typography variant="button"> {backLabel ? backLabel : 'back' }</Typography>
                            </Link>
                        </NextLink>
                    </Box>
                }
            </Box>
        </AppGradientBackground>
    );
};

export default CardLayout;
