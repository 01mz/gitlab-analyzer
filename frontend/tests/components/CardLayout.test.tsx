import {createShallow} from "@material-ui/core/test-utils";
import React from 'react';
import CardLayout from '../../components/CardLayout';



describe("Cardlayout", () =>{
    // @ts-ignore
    let shallow;


    beforeAll(() => {
        shallow = createShallow();
    })

    it("Render CardLayout", () => {
        // @ts-ignore

        const wrapper = shallow(<CardLayout children={} />)
        expect(wrapper).toMatchSnapshot();

    })

})