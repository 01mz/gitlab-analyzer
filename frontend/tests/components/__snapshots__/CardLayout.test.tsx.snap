// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cardlayout Snapshot CardLayout 1`] = `
<CardLayout>
  <AppGradientBackground>
    <Styled(MuiBox)
      alignItems="center"
      className="makeStyles-backgroundGradient-1"
      display="flex"
      height="100vh"
      justifyContent="center"
      width="100vw"
    >
      <div
        className="MuiBox-root MuiBox-root-3 makeStyles-backgroundGradient-1"
      >
        <Styled(MuiBox)
          bgcolor="primary.contrastText"
          borderRadius={45}
          boxShadow={20}
          display="flex"
          flexDirection="column"
          height="60vh"
          justifyContent="center"
          minHeight="450px"
          minWidth="250px"
          padding="20px"
          width="60vw"
        >
          <div
            className="MuiBox-root MuiBox-root-4"
          >
            <Styled(MuiBox)
              id="card-header"
              margin="10px"
              textAlign="center"
            >
              <div
                className="MuiBox-root MuiBox-root-5"
                id="card-header"
              >
                <Image
                  alt="The Gitlab Logo"
                  height="125px"
                  src="/gitlab.svg"
                  width="125px"
                >
                  <div
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "display": "inline-block",
                        "margin": 0,
                        "maxWidth": "100%",
                        "overflow": "hidden",
                        "position": "relative",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "boxSizing": "border-box",
                          "display": "block",
                          "maxWidth": "100%",
                        }
                      }
                    >
                      <img
                        alt=""
                        aria-hidden={true}
                        role="presentation"
                        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI1IiBoZWlnaHQ9IjEyNSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                        style={
                          Object {
                            "border": "none",
                            "display": "block",
                            "margin": 0,
                            "maxWidth": "100%",
                            "padding": 0,
                          }
                        }
                      />
                    </div>
                    <img
                      alt="The Gitlab Logo"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style={
                        Object {
                          "border": "none",
                          "bottom": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": 0,
                          "left": 0,
                          "margin": "auto",
                          "maxHeight": "100%",
                          "maxWidth": "100%",
                          "minHeight": "100%",
                          "minWidth": "100%",
                          "objectFit": undefined,
                          "objectPosition": undefined,
                          "padding": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "visibility": "hidden",
                          "width": 0,
                        }
                      }
                    />
                  </div>
                </Image>
                <WithStyles(ForwardRef(Typography))
                  gutterBottom={true}
                  variant="h6"
                >
                  <ForwardRef(Typography)
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    gutterBottom={true}
                    variant="h6"
                  >
                    <h6
                      className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
                    >
                      GitLab
                      <br />
                      Analyzer
                    </h6>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </Styled(MuiBox)>
            <div />
          </div>
        </Styled(MuiBox)>
      </div>
    </Styled(MuiBox)>
  </AppGradientBackground>
</CardLayout>
`;
